core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/function/aggregate/StatisticCollector.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Eclipse RDF4J contributors.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *******************************************************************************/
+package org.eclipse.rdf4j.query.algebra.evaluation.function.aggregate;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.rdf4j.model.Literal;
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.model.base.CoreDatatype;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+import org.eclipse.rdf4j.query.algebra.evaluation.ValueExprEvaluationException;
+import org.eclipse.rdf4j.query.parser.sparql.aggregate.AggregateCollector;
+
+/**
+ * {@link AggregateCollector} implementation that stores all {@link Literal} values for custom SPARQL aggregate
+ * statistical function processing.
+ *
+ * @author Tomas Kovachev t.kovachev1996@gmail.com
+ */
+public abstract class StatisticCollector implements AggregateCollector {
```

Comment:
Could you mark this as experimental to allow us to change things i the future if we need?